packages/react-native/Libraries/Modal/Modal.js

```diff
@@ -244,6 +273,17 @@ class Modal extends React.Component<Props> {
       this.props.children
     );
 
+    const onDismiss = () => {
+      // OnDismiss is implemented on iOS only.
+      if (Platform.OS === 'ios') {
+        this.setState({isRendering: false}, () => {
+          if (this.props.onDismiss) {
+            this.props.onDismiss();
+          }
+        });
```

Comment:
Question: Wouldn't it be better from an OOT platform standpoint to use early return? If we would check `Platform.OS === 'android'` and return. Then no changes would be needed for macOS to support it.